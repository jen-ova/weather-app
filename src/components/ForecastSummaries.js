import React from "react";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";
import "../styles/ForecastSummaries.css";

const ForecastSummaries = ({ forecasts, onForecastSelect }) => (
  <div className="forecast-summaries">
    {forecasts.map((forecast) => (
      <ForecastSummary
        key={forecast.date}
        date={forecast.date}
        description={forecast.description}
        icon={forecast.icon}
        temperature={forecast.temperature}
        onSelect={onForecastSelect}
      />
    ))}
  </div>
);

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number.isRequired,
      description: PropTypes.string.isRequired,
      icon: PropTypes.string.isRequired,
      temperature: PropTypes.shape({
        min: PropTypes.number,
        max: PropTypes.number,
      }).isRequired,
    })
  ).isRequired,
  onForecastSelect: PropTypes.func.isRequired,
};

export default ForecastSummaries;
