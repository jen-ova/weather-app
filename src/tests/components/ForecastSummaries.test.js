import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 1111111,
      description: "stubDescription1",
      icon: "211",
      temperature: {
        min: 12,
        max: 24,
      },
    },
    {
      date: 2222222,
      description: "stubDescription2",
      icon: "781",
      temperature: {
        min: 7,
        max: 14,
      },
    },
  ];

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
