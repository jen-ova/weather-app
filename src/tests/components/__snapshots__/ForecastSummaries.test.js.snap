// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-thunderstorm"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stubDescription1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu 1st Jan
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-tornado"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        14°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stubDescription2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
